src/app/filter/filter.component.spec.ts

```diff
@@ -0,0 +1,66 @@
+import { ComponentFixture, TestBed } from '@angular/core/testing';
+import { ReactiveFormsModule } from '@angular/forms';
+import { MatIconModule } from '@angular/material/icon';
+import { FilterComponent } from './filter.component';
+
+describe('FilterComponent', () => {
+  let component: FilterComponent;
+  let fixture: ComponentFixture<FilterComponent>;
+
+  const initialState = {
+    currentPage: 1,
+    entriesPerPage: 20,
+    nameFilter: null
+  };
+
+  beforeEach(async () => {
+    await TestBed.configureTestingModule({
+      imports: [MatIconModule, ReactiveFormsModule]
+    }).compileComponents();
+
+    fixture = TestBed.createComponent(FilterComponent);
+    component = fixture.componentInstance;
+  });
+
+  it('should create the component', () => {
+    expect(component).toBeTruthy();
+  });
+
+  it('should have the right initial state', () => {
+    expect(component.filterForm.value).toEqual(initialState);
+  });
+
+  it('should update filter state without name filter', () => {
+    const withoutNameFilter: Required<typeof component.filterForm.value> = {
+      currentPage: 2,
+      entriesPerPage: 50,
+      nameFilter: null
+    };
+
+    component.filterForm.setValue(withoutNameFilter);
+    component.updateFilters();
+
+    expect(component.filterForm.value).toEqual({
+      currentPage: withoutNameFilter.currentPage,
+      entriesPerPage: withoutNameFilter.entriesPerPage,
+      nameFilter: null
+    });
+  });
+
+  it('should update filter state with name filter', () => {
+    const withNameFilter: Required<typeof component.filterForm.value> = {
+      currentPage: 2,
+      entriesPerPage: 50,
+      nameFilter: 'test'
+    };
+
+    component.filterForm.setValue(withNameFilter);
+    component.updateFilters();
+
+    expect(component.filterForm.value).toEqual({
+      currentPage: withNameFilter.currentPage,
+      entriesPerPage: withNameFilter.entriesPerPage,
+      nameFilter: withNameFilter.nameFilter
+    });
```

Comment:
not sure if you can do directly `...toEqual(withNameFilter);`

Reply:
It depends on whether toEqual checks if the properties of the two objects are equal or just that the two objects have the same reference in memory.
The former is the expected behaviour and that is why I have explicitly compared each property.
Still, I have to find out what is the exact behaviour of toEqual.

Reply:
So according to the official documentation toEqual performs a deep equality comparison, thus it could have been used in this case. On the contrary, using toBe would be a mistake as it instead performs a strict equality comparison (so taking references into account).

Source: [Jasmine matchers](https://jasmine.github.io/api/edge/matchers.html)

Reply:
yep, toBe compares the reference value, toEqual compares the properties (deeply)